File: src/main/scala/org/mastercs/bigdata/scala_lang/functional_programming/TheCompanyProcess.scala
package org.mastercs.bigdata.scala_lang.functional_programming

/**
 * @author matemaster
 */
object TheCompanyProcess {

    def cleanNamesByFunctionalProgramming(nameList: List[String]): String = {
        if (nameList == null || nameList.isEmpty) {
            return null
        }
        nameList
                .filter(str => str != null && str.nonEmpty)
                .map(capitalizeString)
                .mkString(",")
    }

    def capitalizeString(str: String): String = {
        str.substring(0, 1).toUpperCase + str.substring(1)
    }
}

File: src/main/java/org/mastercs/bigdata/functional_thinking/TheCompanyProcess.java
package org.mastercs.bigdata.functional_thinking;

import java.util.List;
import java.util.stream.Collectors;

/**
 * @author matemaster
 */
public class TheCompanyProcess {

    public static String cleanNames(List<String> nameList) {
        StringBuilder builder = new StringBuilder();
        for (String s : nameList) {
            if (s != null && s.length() > 1) {
                builder.append(capitalizeString(s)).append(",");
            }
        }
        return builder.substring(0, builder.length() - 1).toString();
    }


    public static String cleanNamesByFunctionalProgramming(List<String> nameList) {
        if (nameList == null || nameList.size() == 0) {
            return null;
        }
        return nameList.stream()
                .filter(str -> str != null && str.isEmpty())
                .map(TheCompanyProcess::capitalizeString)
                .collect(Collectors.joining(","));
    }

    public static String capitalizeString(String str) {
        return str.substring(0, 1).toUpperCase() + str.substring(1);
    }
}
